### ISSUE TITLE
requirements.txt using latest package of markyp-bootstrap4 causes Action to fail

### ISSUE THREAD

**Issue by maishiroma** [rempelj/nunit-html-action#1]
Hello! I first wanted to say thanks for creating this GitHub Action, this has helped out immensely in the project that I'm part of. 

## System

- OS Version: Linux
- Python Version: 3.3.10

## Issue

I noticed that recently (After June 28, 2023) this Action started to error out in my GitHub Action workflow. This was the error that got produced:

```py
 Traceback (most recent call last):
  File "/home/runner/work/_actions/rempelj/nunit-html-action/2d0208d2a33a892c5cfe770925cae8ecee96084d/main.py", line 5, in <module>
    from markyp_bootstrap4.alerts import alert
  File "/usr/local/lib/python3.8/dist-packages/markyp_bootstrap4/__init__.py", line 7, in <module>
    from markyp_html import link, script
  File "/usr/local/lib/python3.8/dist-packages/markyp_html/__init__.py", line 8, in <module>
    from markyp.elements import Element, ElementSequence, StandaloneElement, StringElement
  File "/usr/local/lib/python3.8/dist-packages/markyp/elements.py", line 28, in <module>
    class BaseElement(IElement):
  File "/usr/local/lib/python3.8/dist-packages/markyp/elements.py", line 72, in BaseElement
    def get_element_children(self) -> Optional[Sequence[ElementType | None]]:
TypeError: unsupported operand type(s) for |: '_GenericAlias' and 'NoneType'
```

I tried to replicate this error on my local Linux machine (since this action was ran under Linux) and I was able to confirm that this error is occurring because of an updated package dependency that `markyp_bootstrap4` pulls in. That dependency is:

```
markyp-0.2307.0
```

If we were to go to an older version of that dependency, say, `markyp-0.2306.1`, the script will once again work as intended.

## Proposed Solution

If we changed the `requirements.txt` in this repository to have a pinned package for all of the dependencies `markyp-bootstrap4` uses, this will solve this issue. So instead of just having `markyp-bootstrap4` in the `requirements.txt`, something like this will do the trick:

```
markyp==0.2306.1
markyp-bootstrap4==0.2210.1
markyp-html==0.2306.2
```

**Comment by rjtngit** [comment]
Hey! Thanks very much for reporting this issue. 

I couldn't reproduce your issue. Do you have repro steps or a workflow file I can look at?

I added your suggestion anyways, plus added the python setup action, in this PR: #2 

Let me know if that helps!

**Comment by maishiroma** [comment]
Hello @rjtngit ! Thanks for replying to my issue as well as adding said suggestion to your open PR! 

The workflow I use is in a private repository, but I can share some of the details of it here. Essentially, we run a unit test suite in a separate job and using the outputs of that job, we funnel it into a second job. I have confirmed that we do get valid inputs from our first job. 
- Python Version: `3.10.6`
- OS Version: `ubuntu-20.04 LTS`

Below is a pruned copy of said workflow file that contains the relevant information:

```yml
name: Run Unity Unit Tests

on:
  pull_request: # new changes to PRs to master when changes are made to the assets folder
    types:
      - synchronize
      - opened
      - reopened
    branches:
      - master
    paths:
      - 'Assets/**'

jobs:

result_parser:
    name: Generate Unit Test Report
    needs: unit_tester
    runs-on: ubuntu-20.04
    steps:
      - name: Download Temp Artifact
        id: download
        uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # Pinned to v3.0.1
        with:
          name: ${{ needs.unit_tester.outputs.temp_artifact_name }}
          path: ${{ github.workspace }}
      - name: Create PlayMode HTML Results 
        uses: rempelj/nunit-html-action@2d0208d2a33a892c5cfe770925cae8ecee96084d # Pinned to v1.0.1
        with:
          inputXMLPath: ${{ steps.download.outputs.download-path }}/playmode-results.xml
          outputHtmlPath: ${{ steps.download.outputs.download-path }}/playmode-results.html
      - name: Create EditMode HTML Results
        uses: rempelj/nunit-html-action@2d0208d2a33a892c5cfe770925cae8ecee96084d # Pinned to v1.0.1
        with:
          inputXMLPath: ${{ steps.download.outputs.download-path }}/editmode-results.xml
          outputHtmlPath: ${{ steps.download.outputs.download-path }}/editmode-results.html
      - name: Delete Temp Artifact
        if: always()
        uses: geekyeggo/delete-artifact@54ab544f12cdb7b71613a16a2b5a37a9ade990af # Pinned to v2
        with:
          name: ${{ needs.unit_tester.outputs.temp_artifact_name }}
      - name: Upload Final Artifact
        if: always()
        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # Pinned to tag v3.1.1
        with:
          name: ${{ needs.unit_tester.outputs.final_artifact_name }}
          path: ${{ steps.download.outputs.download-path }}
          retention-days: 14
```

As for reproducing it on my local end, this is what I did:
- Python Version: 3.8.10
- OS Version: Ubuntu 20.04.3 LTS

1. Created a new python venv using `python -m venv NUnut` and sourced it
2. Copied the script from the repository to my local machine (since while looking at the repository, I saw that the action is simply calling a python script)
3. Grabbed some valid NUnit unit test results
4. Installed pre-reqs using requirements.txt that was in repository to my virtual environment (going to call it `requirements_old.txt`)
5. Ran script using said test result

Below is a step by step of the above steps
```
$ python -m venv NUnit

$ source ./NUnit/bin/activate

$ python -m pip list
Package       Version
------------- -------
pip           20.0.2
pkg-resources 0.0.0
setuptools    44.0.0

$ cat requirements_old.txt
markyp-bootstrap4

$ python -m pip install -r requirements_old.txt
Collecting markyp-bootstrap4
  Using cached markyp_bootstrap4-0.2210.1-py3-none-any.whl (39 kB)
Collecting markyp>=0.1910
  Using cached markyp-0.2307.0-py3-none-any.whl (11 kB)
Collecting markyp-html>=0.1910
  Using cached markyp_html-0.2306.2-py3-none-any.whl (12 kB)
Installing collected packages: markyp, markyp-html, markyp-bootstrap4
Successfully installed markyp-0.2307.0 markyp-bootstrap4-0.2210.1 markyp-html-0.2306.2

$ python -m pip freeze
markyp==0.2307.0
markyp-bootstrap4==0.2210.1
markyp-html==0.2306.2

$ python ./convert_nunit_xml_html.py editmode-results.xml
Traceback (most recent call last):
  File "./convert_nunit_xml_html.py", line 5, in <module>
    from markyp_bootstrap4.alerts import alert
  File "/home/mattyshiro14/.python_venvs/NUnit/lib/python3.8/site-packages/markyp_bootstrap4/__init__.py", line 7, in <module>
    from markyp_html import link, script
  File "/home/mattyshiro14/.python_venvs/NUnit/lib/python3.8/site-packages/markyp_html/__init__.py", line 8, in <module>
    from markyp.elements import Element, ElementSequence, StandaloneElement, StringElement
  File "/home/mattyshiro14/.python_venvs/NUnit/lib/python3.8/site-packages/markyp/elements.py", line 28, in <module>
    class BaseElement(IElement):
  File "/home/mattyshiro14/.python_venvs/NUnit/lib/python3.8/site-packages/markyp/elements.py", line 72, in BaseElement
    def get_element_children(self) -> Optional[Sequence[ElementType | None]]:
TypeError: unsupported operand type(s) for |: '_GenericAlias' and 'NoneType'
```

Let me know if there's anything else I can clarify on!

**Comment by rjtngit** [comment]
Thanks. I'll look into it.

In the meantime, could you check if the `fix-deps` branch works in your github action?
`rjtngit/nunit-html-action@fix-deps`

Side note, in this branch you might need to change the permissions of the output folder first:
```
  - name: Make output directory writable
    shell: bash
    run: sudo chmod a+w ${{ steps.download.outputs.download-path }}
```

**Comment by maishiroma** [comment]
No worries! I just went ahead and tried out that new branch you linked, `rjtngit/nunit-html-action@fix-deps` and that resolved the dependency issue.

![image](https://github.com/rjtngit/nunit-html-action/assets/19522250/36c6713a-5171-4d8a-a7e8-f8770d5699fa)

**Comment by rjtngit** [comment]
Thanks for testing!

Released in v1.0.2.
